src/libraries/System.Linq/src/System/Linq/First.cs

```diff
@@ -21,7 +21,7 @@ public static TSource First<TSource>(this IEnumerable<TSource> source)
 
         public static TSource First<TSource>(this IEnumerable<TSource> source, Func<TSource, bool> predicate)
         {
-            TSource? first = source.TryGetFirst(predicate, out bool found);
+            TSource? first = source.TryGetFirst(predicate, default, out bool found);
```

Comment:
Copying `default` comes at a cost and isn't needed in this case. I would recommend using `TryGetFirst` in its original form, and then rewrite the new method in the following way:
```csharp
public static TSource FirstOrDefault<TSource>(this IEnumerable<TSource> source, TSource defaultValue)
{
    TSource? first = source.TryGetFirst(out bool found);
    return found ? first : defaultValue;
}
```

Reply:
Could you clarify this suggestion? I removed the `TryGetFirst(this IEnumerable<TSource> source, out bool default);` overload in a previous commit at your suggestion. Are you wanting me to re-implement this?

```cs
private static TSource? TryGetFirst<TSource>(this IEnumerable<TSource> source, out bool found)
{
    if (source == null)
    {
        ThrowHelper.ThrowArgumentNullException(ExceptionArgument.source);
    }

    if (source is IPartition<TSource> partition)
    {
        return partition.TryGetFirst(out found);
    }

    if (source is IList<TSource> list)
    {
        if (list.Count > 0)
        {
            found = true;
            return list[0];
        }
    }
    else
    {
        using (IEnumerator<TSource> e = source.GetEnumerator())
        {
            if (e.MoveNext())
            {
                found = true;
                return e.Current;
            }
        }
    }

    found = false;
    return default;
}

private static TSource? TryGetFirst<TSource>(this IEnumerable<TSource> source, TSource defaultValue, out bool found)
{
    TSource? first = source.TryGetFirst(out found);
    return found ? first : default;
}
```

Reply:
There's no need to bring back
```csharp
private static TSource? TryGetFirst<TSource>(this IEnumerable<TSource> source, TSource defaultValue, out bool found);
```
if that's what you're asking. My recommendation is to reuse the existing
```csharp
private static TSource? TryGetFirst<TSource>(this IEnumerable<TSource> source, out bool found)
```
method from `main` without modification and layer the new APIs on top of it as suggested in the original comment.

Reply:
```cs
private static TSource? TryGetFirst<TSource>(this IEnumerable<TSource> source, out bool found)
```
This is the method that was removed, so I cannot reuse it without bringing it back. The overload accepting the default value is the only one existing at this time, which is why I ask if you want me to bring it back.

I can remove the `defaultValue` argument, change the return value back to `default`, and implement the null conditional you recommend, either as above or by modifying each of the `FirstOrDefault()` overloads that accept a default value as so:

```cs
public static TSource FirstOrDefault<TSource>(this IEnumerable<TSource> source, TSource defaultValue)
{
    var first = source.TryGetFirst(out bool found);
    return found ? first! : defaultValue;
}
```

Reply:
> This is the method that was removed, so I cannot reuse it without bringing it back. The overload accepting the default value is the only one existing at this time, which is why I ask if you want me to bring it back.
>
> I can remove the defaultValue argument, change the return value back to default, and implement the null conditional you recommend, either as above or by modifying each of the FirstOrDefault() overloads that accept a default value as so:

Yeah exactly. I guess the point I was trying to make is that there's no need to make any changes to [`TryGetFirst` as is defined in `main`](https://github.com/dotnet/runtime/blob/062ae9697dc040d50b128af094f465fb3d8bb0f0/src/libraries/System.Linq/src/System/Linq/First.cs#L39-L73). Same also applies for `TryGetLast`.